### PR TITLE
Update x/net lib

### PR DESCRIPTION
**Pull request by sapk**
Changelog: https://github.com/golang/net/compare/f2499483f923065a842d38eb4c7f1927e6fc6e6d...master

Related: #4817

### CODE REVIEW

File: Gopkg.toml
Comment (techknowlogick) [inline Gopkg.toml:30]: Is it ok to tag a branch, won't `dep ensure` CI check go to latest commit of branch?

Comment (sapk) [reply Gopkg.toml:30]: Dep ensure will follow the Gopkg.lock. The constraint is use for `dep ensure -update` or `dep status` to report new commit.
Exemple with git and sdk:
```
PROJECT                                        CONSTRAINT          VERSION        REVISION  LATEST   PKGS USED
code.gitea.io/git                              branch master       branch master  6ef79e8   f3afe30  1   
code.gitea.io/sdk                              branch master       branch master  021567c   2d00f9f  1   
```
We are not at latest of master currently but `dep ensure` doesn't complain.

**Comment by sapk** [comment]
Note for an other PR: bleve and goth (PR in progress) could be updated to remove the x/net/context vendor since it is a standard package from Go1.7.

**Review by techknowlogick — APPROVED** [review]
LGTM, probably should wait for #5167 before this PR gets merged.

**Review by lafriks — APPROVED** [review]

**Pull request merged**
